### PR TITLE
ENH display grad norm in line search + modified example wLasso

### LINKED ISSUE #60: ENH improve verbose output
Using `verbose=True` in ImplicitForward yields "0 -st iterations over 100",  I am not sure what this is

In my expectations the basic print should be done by the optimizer, every time a new alpha is chosen, and the norm of the gradient for this corresponding alpha? Is this doable easily?

### PR DESCRIPTION
**Pull request by QB3**
Display gradient norm for the outer optimization problem for the line search

closes #60

### CODE REVIEW

**Comment by QB3** [comment]
Do you think to other quantities to display @mathurinm ?

File: sparse_ho/optimizers/line_search.py [hunk 1]
```diff
@@ -115,9 +115,10 @@ def _grad_search(
             value_outer_old = value_outer
 
             if self.verbose:
-                print('value outer', value_outer)
-                # print('grad outer', grad_outer)
-                # print('value of log_alphak', log_alphak)
+                print(
+                    "Iteration %i / %i ||" % (i+1, self.n_outer) +
+                    "Value outer criterion: %f ||" % value_outer +
+                    "norm grad %f" % norm(grad_outer))
```
Comment (agramfort) [inline sparse_ho/optimizers/line_search.py:121]: you should use f-strings these days

File: examples/plot_wlasso.py [hunk 2]
```diff
@@ -89,7 +89,7 @@
 criterion = HeldOutMSE(idx_train, idx_val)
 algo = ImplicitForward()
 monitor = Monitor()
-optimizer = LineSearch(n_outer=20, tol=1e-6)
+optimizer = LineSearch(n_outer=20, tol=1e-6, verbose=True)
 grad_search(algo, criterion, model, optimizer, X, y, log_alpha0, monitor)
```
Comment (mathurinm) [inline examples/plot_wlasso.py:93]: while you're at it, L93 results in a large block of text being displayed on the example (on the doc website), can you assign the result to a variable?

File: sparse_ho/optimizers/line_search.py (hunk 1 above)
Comment (mathurinm) [reply sparse_ho/optimizers/line_search.py:121]: also remove spaces around / after Iteration
norm grad can be in scientific mode with 2 significant digits maybe ? %.2e

**Comment by QB3** [comment]
good for you @agramfort ?

**Review by agramfort — APPROVED** [review]

**Pull request merged**